File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.cs
using System;

public class Solution 
{
    public string solution(string[] my_strings, int[,] parts) 
    {
        string answer = "";
        string temp = "";
        
        string str = "";
        
        int r = 0;
        int start = 0;
        int end = 0;

         while (r<parts.GetLength(0)) {
            start = parts[r,0];
            end = parts[r,1];
            
            temp = my_strings[r];
             
            for (int idx=start; idx<=end; idx++) {
                str += temp[idx].ToString();
            }
            answer += str;
            str = "";   
            r++;
        }     
        return answer;
    }
}

File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> my_strings, vector<vector<int>> parts) 
{
    string sumStr = "";
    for(int i=0; i<my_strings.size(); i++)
    {
        int start = parts[i][0];
        int len = parts[i][1] - parts[i][0] + 1;
        sumStr += my_strings[i].substr(start,len);
    }
    return sumStr;
}